source/m2mtlvserializer.cpp

```diff
@@ -126,8 +126,21 @@ bool M2MTLVSerializer::serialize_resource(const M2MResource *resource, uint8_t *
 {
     bool success = false;
     if(resource->name_id() != -1) {
-        success = serialize_TILV(TYPE_RESOURCE, resource->name_id(),
+        if ( resource->resource_instance_type() == M2MResourceInstance::INTEGER) {
+            int64_t valueInt = resource->get_value_int();
+            uint32_t buf_size;
+            /* max len 8 bytes */
```

Comment:
Ah, I was potentially misreading the spec. Still not totally clear to me whether integer TLVs are intended to be autosized based on value (like CoAP options) or be fixed for each specific resource. The fact that they spell out 8,16,32 and 64 suggests to me at least they're not thinking of autosizing. If autosizing by value I'd expect it to allow any number of bytes from 0-8.

I guess we could handle all cases for writing with 5 distinct types for the 4 fixed sizes and 1 for autosize.